File: CurrencyConverterAPI/Models/Converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CurrencyConverterAPI.Models
{
    public class Converter
    {
        public float amount;
        public float exchangeRate;
        public float result;

        public Converter(float amount, float exchangeRate)
        {
            this.amount = amount;
            this.exchangeRate = exchangeRate;
        }

        public void convert()
        {
            result = amount * exchangeRate;
        }
    }
}

File: ConverterApp/wwwroot/js/converter.js
$(document).ready(function () {

    $('body').on('click', 'button#convert', function (e) {
        console.log('clicked');
        let amount = $('input#amount').val() || $('input#amount').attr('placeholder') || '';
        let exchangeRate = $('input#exchange-rate').val() || $('input#exchange-rate').attr('placeholder') || '';
        if (amount == '' || exchangeRate == '') {
            alert('Please enter an appropriate amount or exchange rate value.');
        } else {
            convert(amount, exchangeRate);
        }
    });

    function convert(amount, exchangeRate) {
        $.ajax({
            url: `https://localhost:44314/api/converter/convert/${amount}/${exchangeRate}`,
            method: 'get',
            success: function (result) {
                console.log(result);
                $('label.result').text("Result: " + result);
            },
            error: function (errors) {
                alert("There was an error. Please try again.");
                console.log(errors);
            }
        });
    }

});